### PR TITLE
feat(query-builder): save and restore inner query state on commit/discard

### PR DESCRIPTION
**Pull request by igdmdimitrov**
Closes #  

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [ ] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by gedinakova** [comment]
@igdmdimitrov Currently, changes, made to conditions on a level where there are other sub-query conditions, are immediately propagated, making discarding the parent sub-query impossible:
1. Starting in this state:
image
2. Make a change to the first most inner condition > Date created after Oct 15, 2024 to Oct 16, 2024
3. Commit the condition
4. Discard the parent > the date created is 16 Oct

Expected result: Date created should be restored to Oct 15.

**Comment by teodosiah** [comment]
An expression without return fields could be committed through UI (this was fixed in the base branch) 
QB_issue15

**Comment by gedinakova** [comment]
@igdmdimitrov 
1. In the dev demo, expand the inner query 
2. Clear the return fields
3. Discard the parent condition

Result: The chip shows no return fields. The tree is correct - the return fields have stayed as initially.
Expected result: The UI should properly reflect the tree state.

**Review by gedinakova — APPROVED** [review]

File: projects/igniteui-angular/src/lib/query-builder/query-builder-tree.component.ts [hunk 1]
```diff
@@ -228,15 +228,15 @@ export class IgxQueryBuilderTreeComponent implements AfterViewInit, OnDestroy {
      */
     @Input()
     public set expressionTree(expressionTree: IExpressionTree) {
-        if (JSON.stringify(expressionTree) !== JSON.stringify(this._expressionTree)) {
+        // if (JSON.stringify(expressionTree) !== JSON.stringify(this._expressionTree)) {
```
Comment (ivanvpetrov) [inline projects/igniteui-angular/src/lib/query-builder/query-builder-tree.component.ts:231]: If we are sure we don't need this, maybe remove the commented lines?

File: projects/igniteui-angular/src/lib/query-builder/query-builder-tree.component.ts [hunk 2]
```diff
@@ -1346,7 +1364,7 @@ export class IgxQueryBuilderTreeComponent implements AfterViewInit, OnDestroy {
                         condition: filteringExpr.condition,
                         conditionName: filteringExpr.condition.name,
                         searchVal: filteringExpr.searchVal,
-                        searchTree: filteringExpr.searchTree, // this.createExpressionGroupItem(filteringExpr.searchTree, groupItem),
```
Comment (ivanvpetrov) [inline projects/igniteui-angular/src/lib/query-builder/query-builder-tree.component.ts:1349]: A left over comment

**Comment by ivanvpetrov** [comment]
@igdmdimitrov 
Found a strange behavior where for some reason the contents of a removed condition are projected to a newly created one

https://github.com/user-attachments/assets/7ce06da8-2e90-40b6-8c52-ad7bf4a77a45

**Review by ivanvpetrov — APPROVED** [review]

**Pull request merged**